src/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -182,7 +182,49 @@ public void UpdatePackageEntry(PackageReference oldEntry, PackageReference newEn
                 matchingNode.ReplaceWith(newPackageNode);
             }
         }
-        
+
+        /// <summary>
+        /// Update a package entry using the original entry as a base if it exists.
+        /// </summary>
+        public void UpdatePackageEntry(XDocument originalConfig, PackageReference newEntry)
```

Comment:
The Method should name UpdateOrAddPackageEntry because if the new entry not exist you are adding her